src/doc/DocHighlightThread.js

```diff
@@ -574,6 +574,28 @@ class DocHighlightThread extends AnnotationThread {
             this.renderAnnotationPopover();
         }
     }
+
+    /**
+     * Fire an event notifying that the comment button has been clicked. Also
+     * show the comment box, and give focus to the text area conatined by it.
+     *
+     * @return {void}
+     */
+    onCommentClick() {
+        this.type = TYPES.highlight_comment;
+        this.state = STATES.pending;
+        this.renderAnnotationPopover();
+    }
+
+    /** @inheritdoc */
+    handleThreadSaveError(error, tempAnnotationID) {
+        if (this.type === TYPES.highlight_comment && this.state === STATES.pending) {
```

Comment:
How does this react to having more than one comment? Will it error out and revert to a plain highlight?

Reply:
the highlight comment will only have a state of pending if it's being toggled from a plain highlight so it should never hit this case if there's multiple comments